### ISSUE TITLE
Add compatibility with NMR Hide by role feature

### ISSUE THREAD

**Issue by Dimensionone** [helgatheviking/nav-menu-roles-woocommerce-memberships#7]
Hi,
can you tell me if I am doing something wrong or there is some with this plugin?

I need to set up:

**Display the menu item only for the logged-in users that don't have the selected membership**

So I setup the menu item and set:

Hide for Logged in users + selected membership plans

But this is not working. The menu items displays, even when the user is logged in and has the selected membership plan.

I tried to test this scenario with just user roles - so I set:

Hide for Logged-in users + selected user role - and it is working fine.

Also, when I select the following conditions:

Hide for Logged-in users + selected user role + selected membership plans - it will stop working, and the menu is displayed, even when the user has the user role + selected membership plans assigned.

The opposite logic - display is just for the logged-in users + selected membership roles working again fine.

**Comment by helgatheviking** [comment]
I don't believe this mini-extension has not been updated to handle the hide by role feature in Nav Menu Roles. I am pretty deep in some other work at the moment, but I could be persuaded to look at this via sponsored development. Let me know if you are interested and I will give some thought to a quote. 

I am also open to a pull request if you are able to resolve it.

**Comment by Dimensionone** [comment]
Hi,

thank you for your email. I tested it and the hide functionality was not
working with the WooCommerce membership roles :(. Sorry, I don't have funds
to cover this fix - creating the sites for some non profit organization.
If you will have time to analyse it and maybe fix it - i am ready to help
with the testing.

Regards,

Viliam
